crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -182,7 +190,10 @@ where
     /// Handler for `eth_newPendingTransactionFilter`
     async fn new_pending_transaction_filter(&self) -> RpcResult<FilterId> {
         trace!(target: "rpc::eth", "Serving eth_newPendingTransactionFilter");
-        self.inner.install_filter(FilterKind::PendingTransaction).await
+        let receiver = self.inner.pool.pending_transactions_listener();
+        let receiver = Arc::new(Mutex::new(receiver));
```

Comment:
hmm, yeah I see why this is needed, and this makes sense because we want to handle draining the receiver without keeping a lock on the filter set, so this makes sense to me.